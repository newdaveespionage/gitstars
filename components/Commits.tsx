import { useState } from 'react'
import CommitItems, { CommitItemsProps } from './CommitItems'

const RepoDetail = (props: CommitItemsProps): JSX.Element => {
  const [showCommits, setShowCommits] = useState<boolean>(false)

  const loadCommits = () => {
    setShowCommits(true)
  }
  return (
    <div>
      {!showCommits && (
        <div onClick={loadCommits} className="">
          Tap to load commits for {props.owner}/{props.name}
        </div>
      )}
      {showCommits && (
        <>
          <h4>Commits from the last 24 hours</h4>
          <CommitItems {...props} />
        </>
      )}
      <style jsx>{``}</style>
    </div>
  )
}

export default RepoDetail
