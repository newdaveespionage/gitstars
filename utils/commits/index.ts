import { RawCommit } from './internalTypes'
import { Commit } from './types'

export const standardizeCommits = (rawCommits: RawCommit[]): Commit[] => {
  const commits = rawCommits.map(
    (rawData): Commit => ({
      id: rawData.node_id,
      name: rawData.commit.committer.name,
      date: rawData.commit.committer.date,
      message: rawData.commit.message,
    })
  )

  return commits
}
