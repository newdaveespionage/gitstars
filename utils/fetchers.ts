const headers = {
  'Content-type': 'application/json',
}

const postConfig = {
  method: 'POST',
  headers,
}

export const repoFetcher = (): Promise<any> =>
  fetch('/api/repos/starred', postConfig)
    .then((res) => res.json())
    .then((json) => json.data)

export const commitsFetcher = (owner: string, name: string): Promise<any> =>
  fetch(`/api/repos/${owner}/${name}`, postConfig)
    .then((res) => res.json())
    .then((json) => json.data)
