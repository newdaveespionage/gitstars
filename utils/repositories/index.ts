import { RawRepository } from './internalTypes'
import { Repository } from './types'

export const standardizeRepositories = (
  rawRepos: RawRepository[]
): Repository[] =>
  rawRepos.map((rawRepo) => ({
    id: rawRepo.id,
    name: rawRepo.name,
    owner: rawRepo.owner.login,
    url: rawRepo.html_url,
    stars: rawRepo.stargazers_count,
    description: rawRepo.description,
  }))
