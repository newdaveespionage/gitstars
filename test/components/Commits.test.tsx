import React from 'react'
import { fireEvent, render } from '../testUtils'
import Commits from '../../components/Commits'
import { commitsFetcher } from '../../utils/fetchers'

jest.mock('../../utils/fetchers', () => ({
  commitsFetcher: jest.fn(),
}))

export const commitsTestProps = {
  owner: 'TestOwner',
  name: 'TestName',
}

describe('CommitItem', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Commits {...commitsTestProps} />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('clicking button triggers alert', () => {
    const { getByText } = render(<Commits {...commitsTestProps} />, {})
    fireEvent.click(getByText('Tap to load commits'))
    expect(commitsFetcher).toHaveBeenCalledWith('TestOwner', 'TestName')
  })
})
