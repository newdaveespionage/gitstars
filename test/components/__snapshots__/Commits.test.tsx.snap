// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitItem matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-3680457408"
  >
    <button
      class="jsx-3680457408 load-cta"
      title="use this button to load commits from the past 24 hours"
    >
      Tap to load commits
    </button>
  </div>
</DocumentFragment>
`;
